azurelinuxagent/common/osutil/redhat.py

```diff
@@ -117,11 +117,38 @@ def set_hostname(self, hostname):
             logger.warn("[{0}] failed, attempting fallback".format(' '.join(hostnamectl_cmd)))
             DefaultOSUtil.set_hostname(self, hostname)
 
+    def get_nm_controlled(self):
+        ifname = self.get_if_name()
+        filepath = "/etc/sysconfig/network-scripts/ifcfg-{0}".format(ifname)
+        nm_controlled_cmd = ['grep', 'NM_CONTROLLED=', filepath]
+        try:
+            result = shellutil.run_command(nm_controlled_cmd, log_error=False).rstrip()
+
+            if result and len(result.split('=')) > 1:
+                value = result.split('=')[1]
+                if "n" in value:
```

Comment:
I feel this is a loose evaluation. what are the possible values? May be value start with 'n' or direct matching something like that would be a strict check

Reply:
could be n, no, y, yes, 'n', 'no', 'y', 'yes', "n", "no", "y", "yes"

Reply:
I can remove any ' or " characters and check for n or no

Reply:
we need to be careful about trailing spaces too when we remove it. If it's complicated, then I'm ok as is.

Reply:
Updated and tested stricter check